core/kernel/tee_misc.c

```diff
@@ -118,3 +120,30 @@ bool core_is_buffer_intersect(paddr_t b, paddr_size_t bl,
 		return false;
 	return true;
 }
+
+void *tee_alloc_cache_aligned(size_t size)
+{
+	void *ptr = NULL;
+	size_t alloc_size = 0;
+	uint32_t cacheline_size = 0;
+
+	/*
+	 * Buffer must be aligned on a cache line:
+	 *  - Buffer start address aligned on a cache line
+	 *  - End of Buffer inside a cache line.
+	 *
+	 * If the size to be allocated is already aligned on a cache line add a
+	 * another cache line.
+	 */
+	cacheline_size = dcache_get_line_size();
```

Comment:
This is the number of bytes in the smallest cache line of all the data caches and unified caches that
are controlled by the PE.

What happens if there is a cache with a larger cache line?

Reply:
A bit hard for me to comment as I do not see other use cases in other CPU's and this was existing code in OP-TEE.

But in this particular use that is demonstrated in this PR we need to co-aligned between (PMU) Xilinx MicroBlaze (32-bit) and (APU) ARM Cortex-A53 (64-bit).

Could our CPU SoC experts comment on this how this should be implemented?

Reply:
One option came to my mind is to actually provide alignment as parameter for operation. Then callee must be able to select best method. Alternatively there can be allocators for different use cases.

Reply:
I was thinking that perhaps dcache version is not so important and switched it to use outer cache variant.

Outer cache variant should cover all cases (even if dcache size would be sufficient).

Introduced new `CFG_OUTER_CACHE_LINE_SHIFT` setting that should be set for platforms. In ATF it seems that size 6 was the maximum and used in many places. But it could be made smaller in some platforms (or larger) if there are such platforms present.

Please check it out.